math/binarymath/checkisnumberpoweroftwo.go

```diff
@@ -0,0 +1,37 @@
+// checkisnumberpoweroftwo.go
+// description: Is the number a power of two
+// details:
+// Checks if a number is a power of two- [Power of two](https://en.wikipedia.org/wiki/Power_of_two)
+// author(s) [red_byte](https://github.com/i-redbyte)
+// see checkisnumberpoweroftwo_test.go
+
+package binarymath
+
+import (
+	"math"
+)
+
+func IsPowerOfTwoBinaryMethod(x int) bool {
+	return x > 0 && (x&(x-1)) == 0
+}
+
+func IsPowerOfTwoUseCycleAndLeftShift(number uint) bool {
+	if number == 0 {
+		return false
+	}
+	for p := uint(1); p > 0; p = p << 1 {
+		if number == p {
+			return true
+		}
+	}
+	return false
+}
+
+func IsPowerOfTwoUseLog(number float64) bool {
```

Comment:
Should it really be in this package - this function doesn't have a single bitwise operation. Unless, by `binary` in the package name you mean the binary operations then that means you can include almost every algorithm.

Reply:
This method was added by me to compare the performance of the binary approach and the conventional

Reply:
but if this method seems to be superfluous to you in this package, maybe it is worth leaving a comment that it was added for performance comparison? Or I can just delete it, it doesn't matter to me